sample-resource-plugin/src/main/java/org/opensearch/sample/resource/actions/transport/CreateResourceTransportAction.java

```diff
@@ -0,0 +1,79 @@
+
+package org.opensearch.sample.resource.actions.transport;
+
+import java.io.IOException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.sample.resource.actions.rest.create.CreateResourceAction;
+import org.opensearch.sample.resource.actions.rest.create.CreateResourceRequest;
+import org.opensearch.sample.resource.actions.rest.create.CreateResourceResponse;
+import org.opensearch.security.spi.resources.Resource;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.client.Client;
+
+import static org.opensearch.common.xcontent.XContentFactory.jsonBuilder;
+import static org.opensearch.sample.utils.Constants.RESOURCE_INDEX_NAME;
+
+public class CreateResourceTransportAction extends HandledTransportAction<CreateResourceRequest, CreateResourceResponse> {
+    private static final Logger log = LogManager.getLogger(CreateResourceTransportAction.class);
+
+    private final TransportService transportService;
+    private final Client nodeClient;
+
+    @Inject
+    public CreateResourceTransportAction(TransportService transportService, ActionFilters actionFilters, Client nodeClient) {
+        super(CreateResourceAction.NAME, transportService, actionFilters, CreateResourceRequest::new);
+        this.transportService = transportService;
+        this.nodeClient = nodeClient;
+    }
+
+    @Override
+    protected void doExecute(Task task, CreateResourceRequest request, ActionListener<CreateResourceResponse> listener) {
+        ThreadContext threadContext = transportService.getThreadPool().getThreadContext();
+        try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
+            createResource(request, listener);
+        } catch (Exception e) {
+            log.info("Failed to create resource", e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void createResource(CreateResourceRequest request, ActionListener<CreateResourceResponse> listener) {
+        Resource sample = request.getResource();
+        try (XContentBuilder builder = jsonBuilder()) {
+            IndexRequest ir = nodeClient.prepareIndex(RESOURCE_INDEX_NAME)
+                .setWaitForActiveShards(1)
+                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                .setSource(sample.toXContent(builder, ToXContent.EMPTY_PARAMS))
+                .request();
+
+            log.info("Index Request: {}", ir.toString());
```

Comment:
Log messages with level `info` are visible by default. Thus, these messages should have value for an average OpenSearch admin. This might be rather a `debug` level?

Reply:
I will got through an update the new added log statements soon.